### PR TITLE
Moved to Julia 0.7

### PR DESCRIPTION
**Pull request by dfdx**
Is it already time to move to Julia 0.7? This PR drops support for all previous versions and fixes errors & deprecations for Julia 0.7-beta. 

Besides deprecations, the only change is addition of `==` method for `IntervalSet` and `DiscreteSet` which otherwise require iteration on Julia 0.7 (I guess on 0.6 they were compared by reference).

### CODE REVIEW

**Comment by CarloLucibello** [comment]
Thanks for doing this! 
This is passing all test on windows nightly, 0.7 beta failure is spurious.  
Please remove Manifest.toml (can also add it to .gitignore).

**Comment by Evizero** [comment]
sorry for my absence, I have not switched to 0.7 yet

i haven't looked in detail but isn't this basically doing the same as #29 ?

**Comment by CarloLucibello** [comment]
#29 is staling since january, so better go with this

**Comment by dfdx** [comment]
Yep, I tried #29 first, but it turned to be easier for me to just do it myself from scratch. 

I've removed Manifest.toml now. 

@Evizero What's your plan for 0.7? If you plan to switch in, say, 1 month, we can put off this PR until then (in Julia 0.7 we can specify branches in dependencies anyway). Otherwise or if you want to keep compatibility with previous versions for a long time I can put Compat back and update the PR.

**Comment by Evizero** [comment]
I really don't know. I have a deadline in 11 days that I am focused on right now. Might be pretty soon after that

**Comment by dfdx** [comment]
Cool! So let's just wait until then.

File: Project.toml
Comment (CarloLucibello) [inline Project.toml]: I'm not sure about this ... maybe remove the version line so that it doesn't conflict with new tags?

Comment (dfdx) [reply Project.toml]: I thought Pkg3 uses this line instead of tags to figure out dependency versions, doesn't it? 

In a couple of my 0.7-ready packages I just set the same number in attobot, but maybe there's a smarter way.

File: appveyor.yml
Comment (joshday) [inline appveyor.yml]: This doesn't block the PR from being merged, but this can be 
```

environment:
  matrix:
  - julia_version: 0.7
  - julia_version: latest
```

**Review by joshday — CHANGES_REQUESTED** [review]
Just getting around to taking a good luck at this PR.  My understanding is that we shouldn't add Project.toml files yet as they will be automatically added by a yet-to-be-made upgrade tool.  If you just remove the Project.toml file I'm ready to merge.

**Comment by dfdx** [comment]
Yeah, I've just seen a discussion about `Project.toml` in Slack, so it seems to be the right way. Removed the file.

AppVeyor is red again, but it looks like an error in the service itself or environment settings. If tests pass locally on Windows, I'd just leave it as is for now.

**Comment by joshday** [comment]
I just pushed a commit that I think should fix appveyor.  We'll see in a moment!

**Review by joshday — APPROVED** [review]

**Comment by joshday** [comment]
That fixed appveyor.  Now I just bumped the StatsBase version to the latest 0.7-only release.  I'll merge once everything is green.  Thanks for tackling this @dfdx!

**Comment by Evizero** [comment]
thanks!

**Pull request merged**
